### ISSUE TITLE
[NBug] Value cannot be null.Parameter name: value

### ISSUE THREAD

**Issue by thimmy687** [gitextensions/gitextensions#6549]
## Current behaviour

Open repository for path "C:\" crash the app

## Expected behaviour
some error msg instead

## Steps to reproduce

Language is german

C:\ is a valid path on my machnie, but no git repository

## Error Details
```
System.ArgumentNullException: Value cannot be null.
Parameter name: value
   at System.String.StartsWith(String value)
   at GitUI.CommandsDialogs.BrowseDialog.GitStatusMonitor.<.ctor>g__WorkTreeChanged|23_3(Object sender, FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.OnChanged(FileSystemEventArgs e)
   at System.IO.FileSystemWatcher.CompletionStatusChanged(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* overlappedPointer)
   at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pOVERLAP)
```

## Environment
- Git Extensions 3.1.0.5766-RC1
- Build 497059b2205086d8a44a33dd29f2cae2b4bfffc2 (Dirty)
- Git 2.19.0.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3394.0
- DPI 96dpi (no scaling)

**Comment by thimmy687** [comment]
If I click in the opening Error window on copy an unhandled exception occured:

![image](https://user-images.githubusercontent.com/7715216/57573037-ba9be600-7422-11e9-8edd-17e9ad3b59f7.png)

**Comment by gerhardol** [comment]
On a slightly different build, I get another exception

## Error Details
```
System.UriFormatException: Ogiltig URI. Det gick inte att bestämma formatet för URI:n.
   vid System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   vid GitCommands.PathUtil.NormalizePath(String path) i F:\dev\gc\gitextensions\GitCommands\PathUtil.cs:rad 134
   vid GitCommands.GitModule..ctor(String workingDir) i F:\dev\gc\gitextensions\GitCommands\Git\GitModule.cs:rad 45
   vid GitUI.CommandsDialogs.BrowseDialog.FormOpenDirectory.LoadClick(Object sender, EventArgs e) i F:\dev\gc\gitextensions\GitUI\CommandsDialogs\BrowseDialog\FormOpenDirectory.cs:rad 100
   vid System.Windows.Forms.Control.OnClick(EventArgs e)
   vid System.Windows.Forms.Button.OnClick(EventArgs e)
   vid System.Windows.Forms.Button.PerformClick()
   vid System.Windows.Forms.Form.ProcessDialogKey(Keys keyData)
   vid System.Windows.Forms.Control.PreProcessMessage(Message& msg)
   vid System.Windows.Forms.Control.PreProcessControlMessageInternal(Control target, Message& msg)
   vid System.Windows.Forms.Application.ThreadContext.PreTranslateMessage(MSG& msg)
```


## Environment
- Git Extensions 3.1.0
- Build f11fad3e20bfc3bbd33377caa22f7ef44da0093f (Dirty)
- Git 2.21.0.windows.1
- Microsoft Windows NT 10.0.17763.0
- .NET Framework 4.7.3324.0
- DPI 96dpi (no scaling)

**Comment by RussKie** [comment]
@thimmy687 could you please provide steps to reproduce - I can't open open non-git repo.

An attempt to open an invalid repo from the dashboard results in the following dialog:
![image](https://user-images.githubusercontent.com/4403806/57691934-ce488600-7688-11e9-8633-974f01c0d63d.png)

An attempt to open a non-git repo via _Start > Recent > C:\_ results in the following dialog: 
![image](https://user-images.githubusercontent.com/4403806/57691975-e9b39100-7688-11e9-9586-47331f7ca5c9.png)

An attempt to open via _Start > Open... > C:\_ does nothing

An attempt to open via _Start > Open... > C:_ results in the error that @gerhardol shown above. That's an easy fix to ignore the error.

**Comment by gerhardol** [comment]
@thimmy687 
Is this fixed?

**Comment by illfated** [comment]
@thimmy687 : You may want to upgrade your Git Extensions to the release version 3.1.0.5877, as well as upgrading Git to 2.21, to make sure you get more up-to-date output (even from the errors).

**Comment by AOrlov** [comment]
Sometimes it's possible to reproduce even with the current build. The problem is in GitStatusMonitor.StartWathchingChanges which might set _gitPath to null since `Path.GetDirectoryName("C:\")` returns null. Then GitSM.WorkTreeChanged throws the exception on line #132 `if (e.FullPath.StartsWith(_gitPath))`.

**Comment by gerhardol** [comment]
@AOrlov 
Can you make a pr?

**Comment by AOrlov** [comment]
@gerhardol 
I apologize, while I was trying to reproduce with:
- Git Extensions 3.1.1.6049
- Build 2f872105b4eb42421d9fdafd68e90a9e27d20d96

looks like it was fixed by #6560. Now I don't see a way to reproduce it again.
